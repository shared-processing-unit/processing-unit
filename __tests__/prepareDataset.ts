import {
    //  createDecisionTreeSample,
    splitDecisionTreeSampleIntoNEstimators,
    ReferenceTpe
} from '../scripts/prepareDataset'
describe('prepare dataset', () => {
    const referenceTable = [
        [
            { refX: 2, value: 0, y: 1 },
            { refX: 0, value: 1, y: 1 },
            { refX: 1, value: 1, y: 0 },
            { refX: 3, value: 1, y: 0 }
        ],
        [
            { refX: 3, value: 0, y: 0 },
            { refX: 1, value: 1, y: 0 },
            { refX: 0, value: 2, y: 1 },
            { refX: 2, value: 3, y: 1 }
        ]
    ] as ReferenceTpe[][] /*
    it('create decisionTree Sample', async () => {
        const matrix = [
            ['YELLOW', 5.4, 'T'],
            ['YELLOW', 3.0, 'F'],
            ['PURPLE', 6.1, 'T'],
            ['YELLOW', 2.5, 'F']
        ]
        const ref = createDecisionTreeSample(matrix as [])
        expect(ref).toStrictEqual(referenceTable)
    })*/
    it('create n random decisionTree samples', () => {
        const mockMath = Object.create(global.Math)
        let counter = 0
        const randomNumbers = [0, 1 / 4, 3 / 4, 0, 0, 0]
        mockMath.random = () => randomNumbers[counter++]

        global.Math = mockMath

        const csvChunks = splitDecisionTreeSampleIntoNEstimators(
            referenceTable,
            3,
            2
        )
        expect(csvChunks).toMatchSnapshot()
    })
})
