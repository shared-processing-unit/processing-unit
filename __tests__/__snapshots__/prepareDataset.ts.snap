// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prepare dataset create n random decisionTree samples 1`] = `
Array [
  Array [
    Array [
      Object {
        "refX": 2,
        "value": 0,
        "y": 1,
      },
      Object {
        "refX": 3,
        "value": 0,
        "y": 0,
      },
    ],
    Array [
      Object {
        "refX": 0,
        "value": 1,
        "y": 1,
      },
      Object {
        "refX": 1,
        "value": 1,
        "y": 0,
      },
    ],
    Array [
      Object {
        "refX": 3,
        "value": 1,
        "y": 0,
      },
      Object {
        "refX": 2,
        "value": 3,
        "y": 1,
      },
    ],
  ],
  Array [
    Array [
      Object {
        "refX": 2,
        "value": 0,
        "y": 1,
      },
      Object {
        "refX": 3,
        "value": 0,
        "y": 0,
      },
    ],
    Array [
      Object {
        "refX": 2,
        "value": 0,
        "y": 1,
      },
      Object {
        "refX": 3,
        "value": 0,
        "y": 0,
      },
    ],
    Array [
      Object {
        "refX": 2,
        "value": 0,
        "y": 1,
      },
      Object {
        "refX": 3,
        "value": 0,
        "y": 0,
      },
    ],
  ],
]
`;
